### ISSUE TITLE
Local answers not used

### ISSUE THREAD

**Issue by ColonelPanicks** [alces-software/metalware#233]
Fresh install of 2017.2.0-rc1, run through of `metal configure domain` and `metal configure local` before running `metal build local` and getting the following:
```
warning: The following returned nil: answer.pri_network_short_hostname
error: Failed to render template: /var/lib/metalware/repo/files/core/base.sh
address family must be specified. Use --trace to view backtrace
```
Trace output:
```
[root@localhost repo]# metal build local --trace
warning: Build file: Template path '/opt/alces/install/scripts/aws.sh' for '/opt/alces/install/scripts/aws.sh' does not exist
warning: Build file: Retrieving 'http://example.com/non-existent-remote-script.sh' gave error '404 Not Found'
warning: Build file: Retrieving 'https://raw.githubusercontent.com/alces-software/metalware-default/master/files/remote-script.sh' gave error '404 Not Found'
warning: Build file: Template path '/var/lib/metalware/repo/files/non-existent-local-script.sh' for 'non-existent-local-script.sh' does not exist
warning: Build file: Retrieving 'http://example.com/non-existent-remote-script.sh' gave error '404 Not Found'
warning: Build file: Retrieving 'https://raw.githubusercontent.com/alces-software/metalware-default/master/files/remote-script.sh' gave error '404 Not Found'
warning: Build file: Template path '/var/lib/metalware/repo/files/non-existent-local-script.sh' for 'non-existent-local-script.sh' does not exist
warning: The following returned nil: answer.pri_network_short_hostname
/opt/metalware/opt/ruby/lib/ruby/2.4.0/ipaddr.rb:479:in `initialize': Failed to render template: /var/lib/metalware/repo/files/core/base.sh (IPAddr::AddressFamilyError)
address family must be specified
	from (erb):65:in `new'
	from (erb):65:in `alces_wrapper_binding'
	from /opt/metalware/opt/ruby/lib/ruby/2.4.0/erb.rb:893:in `eval'
	from /opt/metalware/opt/ruby/lib/ruby/2.4.0/erb.rb:893:in `block in result'
	from /opt/metalware/opt/ruby/lib/ruby/2.4.0/erb.rb:894:in `result'
	from /opt/metalware/src/templating/renderer.rb:62:in `render_erb_template'
	from /opt/metalware/src/templating/renderer.rb:40:in `replace_erb_with_binding'
	from /opt/metalware/src/namespaces/alces.rb:37:in `block in render_erb_template'
	from /opt/metalware/src/namespaces/alces.rb:76:in `run_with_dynamic'
	from /opt/metalware/src/namespaces/alces.rb:35:in `render_erb_template'
	from /opt/metalware/src/namespaces/hash_merger_namespace.rb:44:in `block in template_block'
	from /opt/metalware/src/namespaces/hash_merger_namespace.rb:27:in `render_erb_template'
	from /opt/metalware/src/templater.rb:58:in `render'
	from /opt/metalware/src/templater.rb:77:in `render_to_file'
	from /opt/metalware/src/commands/build.rb:112:in `block (2 levels) in render_build_files'
	from /opt/metalware/src/commands/build.rb:108:in `each'
	from /opt/metalware/src/commands/build.rb:108:in `block in render_build_files'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:35:in `block in each'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:35:in `each'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:35:in `each'
	from /opt/metalware/src/commands/build.rb:107:in `render_build_files'
	from /opt/metalware/src/commands/build.rb:100:in `block in render_build_templates'
	from /opt/metalware/src/commands/build.rb:99:in `each'
	from /opt/metalware/src/commands/build.rb:99:in `render_build_templates'
	from /opt/metalware/src/commands/build.rb:60:in `run'
	from /opt/metalware/src/command_helpers/base_command.rb:41:in `initialize'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:217:in `new'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:217:in `call'
	from /opt/metalware/src/commander_extensions.rb:61:in `call'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:184:in `run'
	from /opt/metalware/src/commander_extensions.rb:49:in `run'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/runner.rb:464:in `run_active_command'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/runner.rb:66:in `run!'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/delegates.rb:15:in `run!'
	from /opt/metalware/src/cli.rb:52:in `run!'
	from /opt/metalware/src/cli.rb:47:in `run'
	from bin/metal:28:in `<main>'
```
I suspect that the render isn't looking for `/var/lib/metalware/answers/local.yml` but might instead be looking for it under the nodes directory?

I've tried specifying a non-default short hostname in the local configure however that doesn't get picked up either.

**Comment by ColonelPanicks** [comment]
It looks like the line between node/local is a bit blurred, when copying the config file I get a different error. Seemingly it's now treating it completely as a node and not as local.
```
[root@localhost repo]# cp ../answers/local.yaml ../answers/nodes/local.yaml
[root@localhost repo]# metal build local
error: Failed to validate answers:
The following question(s) were detected in the answer file but do not appear in 'configure.yaml': ext_network_defined, ext_network_ip, ext_network_gateway. Use --trace to view backtrace
```
Trace:
```
/opt/metalware/src/validation/answer.rb:80:in `data': Failed to validate answers: (Metalware::ValidationFailure)
The following question(s) were detected in the answer file but do not appear in 'configure.yaml': ext_network_defined, ext_network_ip, ext_network_gateway
	from /opt/metalware/src/validation/loader.rb:61:in `answer'
	from /opt/metalware/src/validation/load_save_base.rb:46:in `node_answers'
	from /opt/metalware/src/validation/load_save_base.rb:64:in `section_answers'
	from /opt/metalware/src/hash_mergers/answer.rb:16:in `load_yaml'
	from /opt/metalware/src/hash_mergers/hash_merger.rb:46:in `cached_yaml'
	from /opt/metalware/src/hash_mergers/hash_merger.rb:39:in `block in hash_array'
	from /opt/metalware/src/hash_mergers/hash_merger.rb:35:in `tap'
	from /opt/metalware/src/hash_mergers/hash_merger.rb:35:in `hash_array'
	from /opt/metalware/src/hash_mergers/hash_merger.rb:20:in `merge'
	from /opt/metalware/src/namespaces/hash_merger_namespace.rb:35:in `run_hash_merger'
	from /opt/metalware/src/namespaces/hash_merger_namespace.rb:23:in `answer'
	from /opt/metalware/src/namespaces/hash_merger_namespace.rb:47:in `block in template_block'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:49:in `convert_value'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:53:in `block in convert_value'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:53:in `map'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:53:in `convert_value'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:31:in `[]'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:20:in `method_missing'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:35:in `block in each'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:35:in `each'
	from /opt/metalware/src/hash_mergers/metal_recursive_open_struct.rb:35:in `each'
	from /opt/metalware/src/build_files_retriever.rb:41:in `map'
	from /opt/metalware/src/build_files_retriever.rb:41:in `retrieve'
	from /opt/metalware/src/namespaces/node.rb:84:in `files'
	from /opt/metalware/src/commands/build.rb:107:in `render_build_files'
	from /opt/metalware/src/commands/build.rb:100:in `block in render_build_templates'
	from /opt/metalware/src/commands/build.rb:99:in `each'
	from /opt/metalware/src/commands/build.rb:99:in `render_build_templates'
	from /opt/metalware/src/commands/build.rb:60:in `run'
	from /opt/metalware/src/command_helpers/base_command.rb:41:in `initialize'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:217:in `new'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:217:in `call'
	from /opt/metalware/src/commander_extensions.rb:61:in `call'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:184:in `run'
	from /opt/metalware/src/commander_extensions.rb:49:in `run'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/runner.rb:464:in `run_active_command'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/runner.rb:66:in `run!'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/delegates.rb:15:in `run!'
	from /opt/metalware/src/cli.rb:52:in `run!'
	from /opt/metalware/src/cli.rb:47:in `run'
	from bin/metal:28:in `<main>'
```

**Comment by WilliamMcCumstie** [comment]
I have located the offending section of code that was leading to the bug. Turns out different parts of metalware handles :local different to a :node called 'local'. This distinction is becoming less important with the Namespaces.

As HashMergers relies on the old code, the distinction still needs to be maintained. The fix can be found at: https://github.com/alces-software/metalware/pull/236/commits/e78bd471e7b344ca57ac091dd259416083ad2418
